File: frame/map-frame-project/MapFrameCoreModule/src/main/java/com/grandtech/mapframe/core/editor/gc/IBaseGeoSource.java
package com.grandtech.mapframe.core.editor.gc;

import androidx.annotation.NonNull;

import com.grandtech.mapframe.core.editor.IGcCallBack;
import com.mapbox.geojson.Feature;
import com.mapbox.geojson.FeatureCollection;

import java.util.List;
import java.util.Map;

/**
 * @ClassName BaseGeoSource
 * @Description TODO 操作缓存内的数据
 * @Author: fs
 * @Date: 2021/4/20 9:16
 * @Version 2.0
 */

public interface IBaseGeoSource {


    public String query();

    public String queryCache();

    /**查询Feature根据条件
     * @param con
     * @return List<Feature>
     */
    public List<Feature> queryFeatures(@NonNull Map<String, Object> con);
    /**查询Feature根据条件模糊
     * @param con
     * @return List<Feature>
     */
    public List<Feature> queryFeaturesLike(@NonNull Map<String, Object> con);

    /**查询Feature根据主键
     * @param key
     *  @param value
     * @return
     */
    public Feature queryFeature(@NonNull String key, @NonNull String value);

    /**插入有缓存null值判断
     * @param feature
     * @return
     */
    public Feature insert(Feature feature);

    /**插入无缓存null值判断
     * @param feature
     * @return
     */
    public Feature insertCache(Feature feature);

    /**批量插入有缓存null值判断
     * @param features
     * @return
     */
    public FeatureCollection inserts(List<Feature> features);

    /**批量插入无缓存null值判断 可异步执行
     * @param features
     * @return
     */
    public void insertsCacheSync(List<Feature> features, IGcCallBack iGcCallBack);

    /**删除有缓存null值判断
     * @param feature
     * @return
     */
    public Boolean delete(Feature feature);

    /**
     * 删除无缓存null值判断
     * @param feature
     * @return
     */
    public Boolean deleteCache(Feature feature);

    /**根据条件删除
     * @param con
     * @return List<Feature>
     */
    public Boolean deleteFeatures(@NonNull Map<String, Object> con);

    /**
     * 删除所有有缓存null值判断
     * @return
     */
    public Boolean deleteAll();

    /**更新有缓存null值判断
     * @param feature
     * @return
     */
    public Feature update(Feature feature);

    /**更新无缓存null值判断
     * @param feature
     * @return
     */
    public Feature updateCache(Feature feature);


}

File: frame/map-frame-project/MapFrameCoreModuleV10/src/main/java/com/gradtech/mapframev10/core/offline/geosource/IBaseGeoSource.kt
package com.gradtech.mapframev10.core.offline.geosource

import com.mapbox.geojson.Feature
import com.mapbox.geojson.FeatureCollection
import com.mapbox.maps.extension.style.sources.generated.GeoJsonSource
import org.jetbrains.annotations.NotNull

/**
 * @ClassName IBaseGeoSource
 * @Description TODO 用于离线数据管理的geosource
 * @Author: fs
 * @Date: 2022/9/16 9:55
 * @Version 2.0
 */
interface IBaseGeoSource {


    fun setSourceUrl(@NotNull cachePath: String): IBaseGeoSource

    fun getGeoSource():GeoJsonSource?

    fun initDraw()

    fun query(): String?

    fun queryCache(): String?

    /**查询Feature根据条件
     * @param con
     * @return List<Feature>
    </Feature> */
    fun queryFeatures(@NotNull con: HashMap<String, Any>): List<Feature>

    /**查询Feature根据条件模糊
     * @param con
     * @return List<Feature>
    </Feature> */
    fun queryFeaturesLike(@NotNull con: Map<String, Any>): List<Feature>

    /**查询Feature根据主键
     * @param key
     * @param value
     * @return
     */
    fun queryFeature(@NotNull key: String, @NotNull value: String): Feature?

    /**插入有缓存null值判断
     * @param feature
     * @return
     */
    fun insert(@NotNull feature: Feature): Feature?

    /**插入无缓存null值判断
     * @param feature
     * @return
     */
    fun insertCache(@NotNull feature: Feature): Feature?

    /**批量插入有缓存null值判断
     * @param features
     * @return
     */
    fun inserts(@NotNull features: List<Feature>): FeatureCollection?

    /**批量插入无缓存null值判断 可异步执行
     * @param features
     * @return
     */
    fun insertsCacheUnSync(@NotNull features: List<Feature>, iGcCallBack: IGcCallBack?)

    /**删除有缓存null值判断
     * @param feature
     * @return
     */
    fun delete(@NotNull feature: Feature): Boolean

    /**
     * 删除无缓存null值判断
     * @param feature
     * @return
     */
    fun deleteCache(@NotNull feature: Feature): Boolean

    /**根据条件删除
     * @param con
     * @return List<Feature>
    </Feature> */
    fun deleteFeatures(@NotNull con: HashMap<String, Any>): Boolean

    /**
     * 删除所有有缓存null值判断
     * @return
     */
    fun deleteAll(): Boolean

    /**更新有缓存null值判断
     * @param feature
     * @return
     */
    fun update(feature: Feature?): Boolean

    /**更新无缓存null值判断
     * @param feature
     * @return
     */
    fun updateCache(feature: Feature?): Boolean

    interface IGcCallBack{
        /**
         * gc方法执行后回调，
         */
        fun onGcCallBack(features: List<Feature?>?)

        fun onError(e: Throwable)
    }
}